runtime/moonbase/src/lib.rs

```diff
@@ -1412,31 +1414,18 @@ moonbeam_runtime_common::impl_runtime_apis_plus_common! {
 						}
 					};
 
-					// Calculate the fee that will be taken by pallet transaction payment
-					let fee: u64 = TransactionPayment::compute_fee(
-						xt.encode().len() as u32,
-						&dispatch_info,
-						tip,
-					).saturated_into();
-
-					// Calculate how much gas this effectively uses according to the existing mapping
 					let effective_gas =
 						<Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
 							dispatch_info.weight
 						);
-
-					// Here we calculate an ethereum-style effective gas price using the
-					// current fee of the transaction. Because the weight -> gas conversion is
-					// lossy, we have to handle the case where a very low weight maps to zero gas.
-					let effective_gas_price = if effective_gas > 0 {
-						fee / effective_gas
+					let tip_per_gas = if effective_gas > 0 {
```

Comment:
We should also add an associated rust test for this validation function (if possible)

Reply:
I added some tests, but they're not in good shape. The call to `Executive::validate_transaction` early in the fn fails on an unsigned `UncheckedExtrinsic` except for an Ethereum one, and signing one is proving difficult.

I'll revisit this again with a fresh mind, but we may also want to consider creating a task to do these tests later.

In any case, I agree that they're a good idea...